### ISSUE TITLE
RenderDoc 1.0 - Texture Viewer tab stopped redrawing itself

### ISSUE THREAD

**Issue by MaikelChan** [baldurk/renderdoc#863]
Hi!

I tried sending an error report but it always gave me an SQL error. In case of error it would be nice to be able to go back to the window where you write the description of the error because I lost what I wrote there.

Back to the issue, I was navigating some draw calls in a Vulkan capture and the Texture Viewer stopped redrawing itself. It shows the previously active tab, and changing drawcalls or trying to do anything to update the contents of the tab does nothing. Other tabs work fine.

Here's a screenshot and the error.log:

![renderdoc](https://user-images.githubusercontent.com/7031754/35888259-56094aa6-0b97-11e8-836b-e99487cf508e.png)
[renderdoc_report.zip](https://github.com/baldurk/renderdoc/files/1701082/renderdoc_report.zip)

Keep up the good work!

EDIT: Reloading the capture fixed the problem.

**Comment by baldurk** [comment]
Judging by the log in the report.zip something caused `VK_ERROR_DEVICE_LOST`, but I can't tell what. Do you have a consistent repro of what's causing it to break? If so would you be able to share it with the capture?

I'm also interested to find out why you couldn't send an error report. I tried sending one myself, and it went through fine. I don't see and logs on the server of a failed report. Can you share exactly what you did and what went wrong?

**Comment by MaikelChan** [comment]
I found something consistent:

- Open the attached capture.
- Go to Texture Viewer tab.
- Enable Overlay Wireframe Mesh (this is the key).
- Keep toggling between one of the last draw calls inside Color Pass #1 (for example, the one with EID 700) and the draw call in Color Pass #4.
- It will eventually break more sooner than later. Sometimes it will break on first try.

About the error report, I just went to Help/Send Error Report, entered a description, my email, and checked to upload the capture and the log. I retried several times.

Here's the capture: [http://www.mediafire.com/file/8pxh3r78zfx9mo9/RenderDocv10Error.7z](http://www.mediafire.com/file/8pxh3r78zfx9mo9/RenderDocv10Error.7z)

**Comment by baldurk** [comment]
Thanks, I'll check the capture tomorrow.

For the error report issue, can you send me a [direct email](mailto:baldurk@baldurk.org) to try and diagnose the problem 1-to-1? What you've described is what I've done as well but it works OK for me :disappointed: .

**Comment by baldurk** [comment]
Just a quick update - I've been able to repro this and I'll continue to investigate.

**Comment by baldurk** [comment]
I believe I've fixed the crash - the problem was that when switching between the events with different sized framebuffers, it would momentarily try to render the overlay with the wrong framebuffer size and crash due to mismatched framebuffer/renderarea dimensions.

Since you sent that capture I've changed the serialisation a bit so it doesn't load anymore, but I verified the fix works cherry-picked back a few days. Or else if you're able to capture and test on the next nightly build that would be ideal.
